apps/payments/next/app/[locale]/error.tsx

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use client';
+
+import { useEffect, useState } from 'react';
+import * as Sentry from '@sentry/nextjs';
+import { useRouter, useParams } from 'next/navigation';
+import { CheckoutParams } from './[offeringId]/checkout/[interval]/[cartId]/layout';
+import Image from 'next/image';
+import errorIcon from '@fxa/shared/assets/images/error.svg';
+import { LoadingSpinner } from '@fxa/payments/ui';
+import Link from 'next/link';
+import { Localized } from '@fluent/react';
+import { restartCartAction, getCartAction } from '@fxa/payments/ui/actions';
+import { CartState } from '../../../../../libs/shared/db/mysql/account/src/lib/kysely-types';
+
+interface ErrorParams extends CheckoutParams {
+  [param: string]: any;
+}
+
+export default function Error({
+  error,
+  reset,
+}: {
+  error: Error & { digest?: string };
+  reset: () => void;
+}) {
+  useEffect(() => {
+    Sentry.captureException(error);
+  }, [error]);
+  const router = useRouter();
+  const [loading, setLoading] = useState(false);
+  const { locale, offeringId, interval, cartId } = useParams<ErrorParams>();
+  const hasProductData = locale && offeringId && interval;
+
+  const SUPPORT_URL = process.env.SUPPORT_URL ?? 'https://support.mozilla.org';
+
+  // Reset the view if the cart is in a success state, or restart the cart and redirect
+  async function redirectWithCart() {
+    setLoading(true);
+    try {
+      const cart = await getCartAction(cartId);
+      if (cart.state === CartState.SUCCESS) {
```

Comment:
This should be updated to all states except START. i.e. `cart.state !== CartState.START`.

This does make me think that we should probably have better error handling for PROCESSING, ERROR and SUCCESS states, but that can be done at a later stage. For now just trying to load the page again works as well.

Reply:
I wanted to make sure recreating the cart when it's at its SUCCESS state wouldn't double-charge. If that's not an issue, then I'm happy to make this change

Reply:
Yup. From my understanding the change I'm proposing will reload the current page for cart states of SUCCESS/ERROR/PROCESSING, and will only re-create the cart for START.

Reply:
note: swapping the conditional check (if cart state is "start", then restart the cart; else, call reset() method)